### ISSUE TITLE
Error message - how interpret?

### ISSUE THREAD

**Issue by Walwa** [bcm-uga/TESS3_encho_sen#25]
I have a large matrix, formed by core genome MLST. I have removed all the monomorphic values from the matrix. When I try the command on my large or a shorter matrix : tess3.obj <- tess3(X = test.mat, K = 1:20, ploidy = 1, method = "projected.ls", 
                   coord = coord.strat, openMP.core.num = 30,  rep = 20, max.iteration = 200, 
                   keep = "best", mask = 0, verbose = F)
I get the error message : " Error in CheckX(X, ploidy) :   The maximum value of the genotype matrix (X) cannot be greater than ploidy + 1. Missing data must be encoded as NA."

I have checked and each cell has a single value, but sometimes the value is 3, 4 or 5 as the genotyping is not bi-allelic. Is the issuethat some of my loci have >2 genotypes? Or is it something else?

**Comment by francoio** [comment]
yes, the issue is that some loci have >2 genotypes (for haploid organismd). 

For multiallelic loci, first convert the data with tess2tess3. The function will create one column (or binary marker) for each allele, and the tess3r function will then work fine. Note this how "structure" performs analysis too.

**Comment by Walwa** [comment]
Thank you for your help and for the software.
